src/main/java/ch/njol/skript/sections/SecLoop.java

```diff
@@ -65,32 +66,53 @@
 })
 @Examples({
 	"loop all players:",
-	"\tsend \"Hello %loop-player%!\" to loop-player",
+		"\tsend \"Hello %loop-player%!\" to loop-player",
 	"",
 	"loop items in player's inventory:",
-	"\tif loop-item is dirt:",
-	"\t\tset loop-item to air",
+		"\tif loop-item is dirt:",
+			"\t\tset loop-item to air",
 	"",
 	"loop 10 times:",
-	"\tsend title \"%11 - loop-value%\" and subtitle \"seconds left until the game begins\" to player for 1 second # 10, 9, 8 etc.",
-	"\twait 1 second",
+		"\tsend title \"%11 - loop-value%\" and subtitle \"seconds left until the game begins\" to player for 1 second # 10, 9, 8 etc.",
+		"\twait 1 second",
 	"",
 	"loop {Coins::*}:",
-	"\tset {Coins::%loop-index%} to loop-value + 5 # Same as \"add 5 to {Coins::%loop-index%}\" where loop-index is the uuid of " +
-		"the player and loop-value is the actually coins value such as 200"
+		"\tset {Coins::%loop-index%} to loop-value + 5 # Same as \"add 5 to {Coins::%loop-index%}\" where loop-index is the uuid of " +
+			"the player and loop-value is the actually coins value such as 200",
+	"",
+	"loop {data::*} as {_data}: # {_data} = loop-value",
+		"\tif {_data} > 500:",
+			"\t\tset {_pass} to true",
+	"",
+	"loop {balances::*} as {_uuid} and {_balance}:",
+		"\tif {_balance} < 200:",
+			"\t\tsend \"No enough money!\" to {_uuid} parsed as a player"
 })
-@Since("1.0")
+@Since("1.0, INSERT VERSION (index/value reference)")
 public class SecLoop extends LoopSection {
 
 	static {
-		Skript.registerSection(SecLoop.class, "loop %objects%");
+		Skript.registerSection(SecLoop.class,
+			"loop %objects% with index as %-~object%", // index
+			"loop %objects% [with index] as %-~object%[ and|,] [with value] %-~object%", // index, value
```

Comment:
This pattern could put `as %object% %object%` together which isn't okay, there must be some kind of joiner.

Reply:
Good catch! fixed \:D